### PR TITLE
PLT-2555/PLT-5009/PLT-5225 Changed system messages to be rendered by the client

### PR DESCRIPTION
**Pull request by hmhealey**
The main thing that this gives us is that system messages will now always be in the client's language, regardless of how the server is set up. It also makes some formatting changes that are easy to do now that the client provides special handling for these system messages.

Something important to note is that these messages are still translated on the server into the server language so that notifications still fire properly and integrations will still have something to work with. Because of this, the UI strings on the client use the same key as the ones on the server even though their format is slightly different.

This PR is based upon https://github.com/mattermost/platform/pull/5193. The changes specific to this PR are https://github.com/mattermost/platform/compare/d395466...051f23c

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-2555
https://mattermost.atlassian.net/browse/PLT-5009
https://mattermost.atlassian.net/browse/PLT-5225

#### Checklist
- Has UI changes
- Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/platform/blob/master/i18n/en.json) and [.../webapp/i18n/en.json](https://github.com/mattermost/platform/tree/master/webapp/i18n/en.json)) updates

### CODE REVIEW

File: webapp/components/post_view/components/system_message_helpers.jsx [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+const systemMessageRenderers = {
+    [PostTypes.JOIN_CHANNEL]: renderJoinChannelMessage,
+    [PostTypes.LEAVE_CHANNEL]: renderLeaveChannelMessage,
+    [PostTypes.ADD_TO_CHANNEL]: renderAddToChannelMessage,
+    [PostTypes.REMOVE_FROM_CHANNEL]: renderRemoveFromChannelMessage,
+    [PostTypes.HEADER_CHANGE]: renderHeaderChangeMessage,
+    [PostTypes.DISPLAYNAME_CHANGE]: renderDisplayNameChangeMessage,
+    [PostTypes.PURPOSE_CHANGE]: renderPurposeChangeMessage,
+    [PostTypes.CHANNEL_DELETED]: renderChannelDeletedMessage
+};
```
Comment (jwilander) [inline webapp/components/post_view/components/system_message_helpers.jsx]: Maybe this is the right method, but is there no way to have a general solution so we don't have to write special case code for each new system message? Can we not just pass in the localization id, drop in the whole post.props bag (or a special part of it constructed on the server) into the values prop?

File: webapp/i18n/fr.json [hunk 2]
```diff
@@ -968,6 +968,18 @@
   "analytics.team.title": "Statistiques d'équipe pour {team}",
   "analytics.team.totalPosts": "Nombre de messages",
   "analytics.team.totalUsers": "Nombre d'utilisateurs",
+  "api.channel.add_member.added": "{addedUsername} a été ajouté au canal par {username}",
```
Comment (coreyhulen) [inline webapp/i18n/fr.json]: Why only french?  should this be handled by Pootle.

File: webapp/components/post_view/components/system_message_helpers.jsx (hunk 1 above)
Comment (hmhealey) [inline webapp/components/post_view/components/system_message_helpers.jsx]: That might work for a few of them, but most of the functions have some sort of special behaviour like rendering individual fields with/without markdown or using different strings when the header is added/removed. Maybe we can do that when we add some more of these, but I don't think we need to right now.

File: webapp/i18n/fr.json (hunk 2 above)
Comment (hmhealey) [inline webapp/i18n/fr.json]: I just copied the French ones from the server so I could test out the client-side rendering.

@enahum Is it okay if these are included in this PR or should I only keep the English ones?

Comment (enahum) [inline webapp/i18n/fr.json]: Its fine they will be re-submitted with the translations PR, we normally ask for the english only as it prevents duplicate work from the translators

**Pull request merged**